tiledb/api/src/tests/mod.rs

```diff
@@ -0,0 +1,29 @@
+pub mod examples;
+
+pub mod prelude {
+    pub mod array {
```

Comment:
This prelude pattern seems odd to me. Googling I'm not really finding any specific "this is an idiomatic prelude" type guidance so there's nothing I can point to specifically except to note that every prelude I've seen before was intended to be `use thing::prelude::*` so having `use thing::prelude::other_thing::*` seems a bit awkward. `use thing::other_thing::prelude::*` or just removing the arra/query/strategy sub-modules altogether seems better.

But given I can't find a single example "most folks do it this way" I'll not request anything specific and will just consider this food for thought.

Reply:
Yeah that's fair. I figured it would be a start, and it's internal anyway due to being inside `tests` module.

Having `prelude::*` be the trailing part of the path sounds nice so I'll re-organize this a bit.